### PR TITLE
mgmt-2411 error response formatting

### PR DESCRIPTION
**Pull request by RazRegev**

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -1719,8 +1719,8 @@ func returnRegisterHostTransitionError(
 	if isRegisterHostForbiddenDueWrongBootOrder(err) {
 		return installer.NewRegisterHostForbidden().WithPayload(
 			&models.InfraError{
-				Code:    swag.Int32(http.StatusForbidden),
-				Message: swag.String(err.Error()),
+				Code:    http.StatusForbidden,
```
Comment (filanov) [inline internal/bminventory/inventory.go:1722]: Why not change it into `common.NewApiError(http.StatusForbidden,err)`?

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -1647,10 +1647,7 @@ func (b *bareMetalInventory) RegisterHost(ctx context.Context, params installer.
 			b.eventsHandler.AddEvent(ctx, params.ClusterID, params.NewHostParams.HostID, models.EventSeverityError,
 				"Failed to register host: cluster cannot accept new hosts in its current state", time.Now())
 			return installer.NewRegisterHostForbidden().
```
Comment (filanov) [inline internal/bminventory/inventory.go:1649]: Maybe i wasn't clear
```suggestion
			return common.NewApiError(http.StatusForbidden, err)
```

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -1646,11 +1646,8 @@ func (b *bareMetalInventory) RegisterHost(ctx context.Context, params installer.
 				params.NewHostParams.HostID, params.ClusterID.String(), err.Error())
 			b.eventsHandler.AddEvent(ctx, params.ClusterID, params.NewHostParams.HostID, models.EventSeverityError,
 				"Failed to register host: cluster cannot accept new hosts in its current state", time.Now())
-			return installer.NewRegisterHostForbidden().
-				WithPayload(&models.InfraError{
-					Code:    swag.Int32(http.StatusForbidden),
-					Message: swag.String(err.Error()),
-				})
+			return common.NewApiError(http.StatusForbidden, err)
+				WithPayload(common.GenerateInfraError(http.StatusForbidden, err))
```
Comment (filanov) [inline internal/bminventory/inventory.go:1650]: you don't really need a `WithPayload` part.

**Comment by RazRegev** [comment]
will break backward compatibility

**Pull request closed without merging**